build/patch-charts.sh

```diff
@@ -333,10 +333,10 @@ function patchSidecarInjector() {
         - SETUID
   }' ${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml
 
-  sed_wrap -i -e 's/runAsUser: 1337/runAsUser: {{ .ProxyUID }}/g' ${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml
-  sed_wrap -i -e 's/runAsGroup: 1337/runAsGroup: {{ .ProxyUID }}/g' ${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml
-  sed_wrap -i -e 's/fsGroup: 1337/fsGroup: {{ .ProxyGID }}/g' ${HELM_DIR}/istio-control/istio-discovery/files/injection-template.yaml
-  sed_wrap -i -e 's/fsGroup: 1337/fsGroup: {{ .ProxyGID }}/g' ${HELM_DIR}/istio-control/istio-discovery/files/gateway-injection-template.yaml
+  sed_wrap -i -e 's/runAsUser: 1337/runAsUser: {{ .ProxyUID }}/g' \
+    -e 's/runAsGroup: 1337/{{ if .ProxyUID }}\n      runAsGroup: {{ .ProxyUID }}\n      {{ end }}/g' \
```

Comment:
Instead of adding a specific number of spaces, it would be better to capture the whitespace in the match and use it in the replace (something like `s/\s*runAsGroup/`)